diff --git a/tests/test-utils.c b/tests/test-utils.c
index bd1bab05..1276b1e5 100644
--- a/tests/test-utils.c
+++ b/tests/test-utils.c
@@ -223,6 +223,15 @@ apache_cleanup (void)
 
 #endif /* HAVE_APACHE */
 
+/* Fix g_canonicalize_filename not found in glib-2.0 v2.56.1 */
+#if !GLIB_CHECK_VERSION(2,58,0)
+#ifndef _XOPEN_SOURCE
+#define _XOPEN_SOURCE 500
+#endif
+#include <limits.h>
+#include <stdlib.h>
+#endif
+
 SoupSession *
 soup_test_session_new (GType type, ...)
 {
@@ -242,10 +251,19 @@ soup_test_session_new (GType type, ...)
 		char *abs_cafile;
 
 		cafile = g_test_build_filename (G_TEST_DIST, "test-cert.pem", NULL);
+#if GLIB_CHECK_VERSION(2,58,0)		
 		abs_cafile = g_canonicalize_filename (cafile, NULL);
+#else
+		abs_cafile = realpath(cafile, NULL);
+#endif		
 		g_free (cafile);
 		tlsdb = g_tls_file_database_new (abs_cafile, &error);
+
+#if GLIB_CHECK_VERSION(2,58,0)		
 		g_free (abs_cafile);
+#else
+		free (abs_cafile);
+#endif		
 		if (error) {
 			if (g_strcmp0 (g_getenv ("GIO_USE_TLS"), "dummy") == 0)
 				g_clear_error (&error);
