diff --git a/configure.ac b/configure.ac
index afd089c..1d5fa31 100644
--- a/configure.ac
+++ b/configure.ac
@@ -157,9 +157,9 @@ dnl *** gtk-doc ***
 dnl ***************
 # gtkdocize greps for ^GTK_DOC_CHECK and parses it, so you need to have
 # it on it's own line.
-m4_ifdef([GTK_DOC_CHECK], [
-GTK_DOC_CHECK([1.20], [--flavour no-tmpl])
-],[])
+#m4_ifdef([GTK_DOC_CHECK], [
+#GTK_DOC_CHECK([1.20], [--flavour no-tmpl])
+#],[])
 
 dnl *****************************
 dnl *** gobject-introspection ***
