diff --git a/libsoup/soup-socket.c b/libsoup/soup-socket.c
index baa9290..dac29c8 100644
--- a/libsoup/soup-socket.c
+++ b/libsoup/soup-socket.c
@@ -716,8 +716,13 @@ new_socket_client (SoupSocket *sock)
 	if (priv->timeout)
 		g_socket_client_set_timeout (client, priv->timeout);
 
-	if (priv->local_addr)
-		g_socket_client_set_local_address (client, soup_address_get_gsockaddr (priv->local_addr));
+	if (priv->local_addr) {
+		GSocketAddress *addr;
+
+		addr = soup_address_get_gsockaddr (priv->local_addr);
+		g_socket_client_set_local_address (client, addr);
+		g_object_unref (addr);
+	}
 
 	return client;
 }
